### PR TITLE
feat!: store `wait_time`/`handle_time` in milliseconds

### PR DESCRIPTION
**Pull request by kbond**
Ref: https://github.com/zenstruck/messenger-monitor-bundle/pull/105#issuecomment-2474912832

### CODE REVIEW

**Comment by kbond** [comment]
@bendavies, @bobvandevijver, mind giving this a review? :heart:

**Comment by kbond** [comment]
Should I be using `bigint`? If I'm calculating correct, we can only store ~24 days in milliseconds with a signed int with MySQL.

_(pretty sure the answer is yes so I updated to use `bigint`)_

**Comment by bobvandevijver** [comment]
`bigint` is the way to go indeed, lgtm 👍🏻

**Comment by bendavies** [comment]
LGTM, but it's still displaying the times in seconds in the UI?

**Comment by bendavies** [comment]
looks like you can use `DateTimeImmutable::format('Uv')` to get millseconds, instead of messing with `format('U.u') * 1000` ?

https://3v4l.org/ME1X7

**Comment by kbond** [comment]
> looks like you can use DateTimeImmutable::format('Uv') to get millseconds

Excellent find, thanks!

> LGTM, but it's still displaying the times in seconds in the UI?

The views show them as human readable durations (ie "4 seconds") but if you hover over, it shows the partial seconds (ie 4.032s).

I decided to have the objects that return durations (ie `ProcessedMessage::timeToHandle()`) return seconds as a `float` but what do you think? Should they return the milliseconds as an `int` then convert to seconds in the UI?

**Comment by bendavies** [comment]
Personally, as most of my messages take less than one second, I would prefer to see `xxx ms` rather than `0s` and have to hover.

**Comment by bendavies** [comment]
We could display 0.xxx if less than 1 second?

**Comment by kbond** [comment]
Ok, I'll see what I can do.

**Comment by bobvandevijver** [comment]
Maybe displaying ms if smaller than 5 or 10 seconds, and using s for everything above is nice?

File: templates/components/snapshot.html.twig [hunk 1]
```diff
@@ -35,21 +35,13 @@
...
             </tr>
             <tr>
                 <th class="text-end">Average Handling Time</th>
                 <td>
-                    {% if helper.canFormatDuration %}
-                        <abbr title="{{ snapshot.averageHandlingTime|round }}s">{{ helper.timeFormatter.formatDuration(snapshot.averageHandlingTime) }}</abbr>
-                    {% else %}
-                        {{ snapshot.averageHandlingTime|round }}s
-                    {% endif %}
+                    <abbr title="{{ snapshot.averageHandlingTime|round }}s">{{ helper.formatDuration(snapshot.averageHandlingTime) }}</abbr>
```
Comment (bendavies) [inline templates/components/snapshot.html.twig:44]: bug here, 
`snapshot.averageHandlingTime|round }}s` is showing millseconds with `s` suffix.

File: templates/detail.html.twig [hunk 2]
```diff
@@ -66,51 +66,31 @@
                         <tr>
                             <th class="text-end text-nowrap">Time in Queue</th>
                             <td>
-                                {% if helper.canFormatDuration %}
-                                    <abbr title="{{ message.timeInQueue }}s">{{ helper.timeFormatter.formatDuration(message.timeInQueue) }}</abbr>
-                                {% else %}
-                                    {{ message.timeInQueue }}s
-                                {% endif %}
+                                <abbr title="{{ message.timeInQueue }}s">{{ helper.formatDuration(message.timeInQueue) }}</abbr>
```
Comment (bendavies) [inline templates/detail.html.twig:69]: and here

File: templates/schedule.html.twig [hunk 3]
```diff
@@ -123,19 +123,15 @@
...
                                     <em class="text-secondary">Never</em>
                                 {% endif %}
                             </td>
                             <td class="text-center">
-                                {% if message and helper.canFormatDuration %}
-                                    <abbr title="{{ message.timeToHandle }}s">{{ helper.timeFormatter.formatDuration(message.timeToHandle) }}</abbr>
-                                {% elseif message %}
-                                    {{ message.timeToHandle }}s
+                                {% if message %}
+                                    <abbr title="{{ message.timeToHandle }}s">{{ helper.formatDuration(message.timeToHandle) }}</abbr>
```
Comment (bendavies) [inline templates/schedule.html.twig:134]: and here

File: templates/statistics.html.twig [hunk 4]
```diff
@@ -82,18 +82,10 @@
                                 {% endif %}
                             </td>
                             <td>
-                                {% if helper.canFormatDuration %}
-                                    <abbr title="{{ metric.averageWaitTime|round }}s">{{ helper.timeFormatter.formatDuration(metric.averageWaitTime) }}</abbr>
-                                {% else %}
-                                    {{ metric.averageWaitTime|round }}s
-                                {% endif %}
+                                <abbr title="{{ metric.averageWaitTime|round }}s">{{ helper.formatDuration(metric.averageWaitTime) }}</abbr>
```
Comment (bendavies) [inline templates/statistics.html.twig:85]: here

File: templates/statistics.html.twig [hunk 5]
```diff
@@ -82,18 +82,10 @@
...
                             </td>
                             <td>
-                                {% if helper.canFormatDuration %}
-                                    <abbr title="{{ metric.averageHandlingTime|round }}s">{{ helper.timeFormatter.formatDuration(metric.averageHandlingTime) }}</abbr>
-                                {% else %}
-                                    {{ metric.averageHandlingTime|round }}s
-                                {% endif %}
+                                <abbr title="{{ metric.averageHandlingTime|round }}s">{{ helper.formatDuration(metric.averageHandlingTime) }}</abbr>
```
Comment (bendavies) [inline templates/statistics.html.twig:88]: here

**Comment by kbond** [comment]
Thanks @bendavies! Fixed in #128.

**Pull request merged**